File: challenges/Custom_Msgs_Srvs/src/battery_node.cpp
#include "includes/BatteryInfo.hpp"

int	main(int argc, char **argv)
{
	Battery info;

	ros::init(argc, argv, "Battery_Node");
	ros::NodeHandle nh;
	info.pub = nh.advertise<challenge_msgs::BatteryStatus>("/battery_node", 10);

	while (ros::ok()) {
		sleep(3);
		info.battery_info(100, true, "Battery full");
		info.pub.publish(info.msg);
		sleep(3);
		info.battery_info(50, true, "Half charge");
		info.pub.publish(info.msg);
		sleep(3);
		info.battery_info(0, false, "Battery depleted");
		info.pub.publish(info.msg);
	}

}

File: basics/Custom_Msgs_Srvs/script/battery_node.py
#!/usr/bin/env python3

import rospy
from challenge_msgs.msg import BatteryStatus

def statusPublisher(percent, status, warn, msg, pub):
    msg.battery_percentage = percent
    msg.battery_on = status
    msg.warn_message = warn
    pub.publish(msg)

if __name__ == "__main__":
    rospy.init_node("Battery_Node")
    
    pub = rospy.Publisher("/battery_node", BatteryStatus, queue_size = 10)


    while not rospy.is_shutdown():
        msg = BatteryStatus()
        rospy.sleep(3)
        statusPublisher(100, True, "Battery full", msg, pub)
        rospy.sleep(3)
        statusPublisher(50, True, "Half charge", msg, pub)
        rospy.sleep(3)
        statusPublisher(0, False, "Battery depleted", msg, pub)
